command/monitor/monitor.go

```diff
@@ -0,0 +1,143 @@
+package monitor
+
+import (
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+)
+
+// Monitor provides a mechanism to stream logs using go-hclog
+// InterceptLogger and SinkAdapter. It allows streaming of logs
+// at a different log level than what is set on the logger.
+type Monitor interface {
+	// Start returns a channel of log messages which are sent
+	// ever time a log message occurs
+	Start() <-chan []byte
+
+	// Stop de-registers the sink from the InterceptLogger
+	// and closes the log channels
+	Stop()
+}
+
+// monitor implements the Monitor interface
+type monitor struct {
+	sink log.SinkAdapter
+
+	// logger is the logger we will be monitoring
+	logger log.InterceptLogger
+
+	// logCh is a buffered chan where we send logs when streaming
+	logCh chan []byte
+
+	// doneCh coordinates the shutdown of logCh
+	doneCh chan struct{}
+
+	// droppedCount is the current count of messages
+	// that were dropped from the logCh buffer.
+	droppedCount uint64
+	bufSize      int
+	// dropCheckInterval is the amount of time we should
+	// wait to check for dropped messages. Defaults
+	// to 3 seconds
+	dropCheckInterval time.Duration
+}
+
+// NewMonitor creates a new Monitor. Start must be called in order to actually start
+// streaming logs
+func NewMonitor(buf int, logger log.InterceptLogger, opts *log.LoggerOptions) Monitor {
+	return newMonitor(buf, logger, opts)
+}
+
+func newMonitor(buf int, logger log.InterceptLogger, opts *log.LoggerOptions) *monitor {
+	sw := &monitor{
+		logger:          logger,
+		logCh:           make(chan []byte, buf),
+		doneCh:          make(chan struct{}, 1),
```

Comment:
doneCh doesn't need buffering does it?  I think we never write to it, only close it.

Reply:
You're right.